compiler/src/iree/compiler/Codegen/Transforms/Transforms.cpp

```diff
@@ -842,33 +873,64 @@ struct FoldExpandShapeIntoInterfaceTensorStore
 ///       tensor<3x?x?x96xf32> ->
 ///       !flow.dispatch.tensor<writeonly:tensor<3x?x?x96xf32>>{%d0, %d1}
 ///
-/// TODO: This handles full slices. The pattern below
-/// (`FoldCollapseShapeIntoTensorInsertSlice`) handles cases where the slic is
+/// TODO: This handles full slices (along collapsed dims). The pattern below
```

Comment:
We only check the strides. Are we missing the check for offsets? I left a comment in lit tests, and I wonder if that is intended.

Reply:
We check both to make sure it's a full slice along the non-collapsed dimensions. The criteria for 'full slice' is to have a matching size and offset = 0.

Reply:
Sorry, you are right on this! I over-indexed on the lit test example and I missed that I reviewed the snippet below. 🤦